packages/app/features/auth/account-recovery/account-recovery.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useState, useCallback } from 'react'
+import RecoverWithEOA from 'app/features/auth/account-recovery/eoa/RecoverWithEOA'
+import { Stack, XStack, YStack, Text, Button, ButtonText } from '@my/ui'
+import type {
+  ChallengeResponse,
+  VerifyChallengeRequest,
+  ErrorWithMessage,
+} from '@my/api/src/routers/account-recovery/types'
+import { RecoveryOptions } from '@my/api/src/routers/account-recovery/types'
+import { api } from 'app/utils/api'
+import { IconError, IconRefresh } from 'app/components/icons'
+import { useRouter } from 'solito/router'
+import RecoverWithPasskey from 'app/features/auth/account-recovery/passkey/RecoverWithPasskey'
+
+interface Props {
+  onClose?: () => void
+}
+
+export enum SignState {
+  NOT_COMPLETE = 0,
+  IN_PROGRESS = 1,
+  FAILED = 2,
+}
+
+export default function AccountRecovery(props: Props) {
+  const [error, setError] = useState<ErrorWithMessage>()
+  // TODO: frontend: implement signing loading state
+  const [signState, setSignState] = useState<SignState>(SignState.NOT_COMPLETE)
+  const [challengeData, setChallengeData] = useState<ChallengeResponse>()
+  const { mutateAsync: getChallengeMutateAsync } = api.challenge.getChallenge.useMutation({
+    retry: false,
+  })
+  const { mutateAsync: validateSignatureMutateAsync } = api.challenge.validateSignature.useMutation(
+    { retry: false }
+  )
+  const router = useRouter()
+
+  useEffect(() => {
+    const loadChallenge = async () => {
+      await getChallengeMutateAsync()
+        .then((challengeResponse) => {
+          setChallengeData(challengeResponse as ChallengeResponse)
+        })
+        .catch((err) => {
+          setError(err)
+        })
+    }
+    loadChallenge()
+  }, [getChallengeMutateAsync])
+
+  const onSignSuccess = useCallback(
+    async (verifyChallengeRequest: VerifyChallengeRequest) => {
+      await validateSignatureMutateAsync({
+        ...verifyChallengeRequest,
+      })
+        .then(() => {
+          // JWT is set via Set-Cookie header
+          router.push('/')
+        })
+        .catch((err) => {
+          setError(err)
+        })
+    },
+    [validateSignatureMutateAsync, router]
+  )
+
+  const onSignError = () => {
+    setError({
```

Comment:
need to show the user the error.


```suggestion
    const onSignError = (e: SignMessageErrorType) => {
    setError({
      message: `Failed to sign challenge. ${e.message.split('.')[0]}`,
    })
  }
```

Reply:
what it looks like for me on first try trying to recover EOA with this suggestion.

<img width="477" alt="Screenshot 2024-06-03 at 08 09 23" src="https://github.com/0xsend/sendapp/assets/95193764/f020ee39-f596-447d-a2ce-8918883009d8">